File: app/src/main/java/org/schulcloud/mobile/data/model/RealmString.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmModel;
import io.realm.annotations.RealmClass;

@RealmClass
public class RealmString implements RealmModel {
    public String value;

    public RealmString() {
    }
    public RealmString(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
    public void setValue(String value) {
        this.value = value;
    }
}

File: app/src/main/java/org/schulcloud/mobile/models/base/RealmString.kt
package org.schulcloud.mobile.models.base

import io.realm.RealmModel
import io.realm.annotations.RealmClass


@RealmClass
open class RealmString @JvmOverloads constructor(var value: String = "") : RealmModel {
    override fun equals(other: Any?): Boolean {
        return when (other) {
            this === other -> true
            is String -> value == other
            is RealmString -> value == other.value
            else -> false
        }
    }

    override fun hashCode() = value.hashCode()
}
